huaweicloud/services/rds/data_source_huaweicloud_rds_mysql_accounts.go

```diff
@@ -0,0 +1,177 @@
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceRdsMysqlAccountsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		listMysqlAccountsHttpUrl = "v3/{project_id}/instances/{instance_id}/db_user/detail?page=1&limit=100"
+		listMysqAccountsProduct  = "rds"
+	)
+	listMysqlAccountsClient, err := cfg.NewServiceClient(listMysqAccountsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	listMysqlAccountsPath := listMysqlAccountsClient.Endpoint + listMysqlAccountsHttpUrl
+	listMysqlAccountsPath = strings.ReplaceAll(listMysqlAccountsPath, "{project_id}", listMysqlAccountsClient.ProjectID)
+	listMysqlAccountsPath = strings.ReplaceAll(listMysqlAccountsPath, "{instance_id}", fmt.Sprintf("%v", d.Get("instance_id")))
+
+	listMysqlAccountsResp, err := pagination.ListAllItems(
+		listMysqlAccountsClient,
+		"page",
+		listMysqlAccountsPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS Mysql accounts")
+	}
+
+	listMysqlAccountsRespJson, err := json.Marshal(listMysqlAccountsResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var listMysqlAccountsRespBody interface{}
+	err = json.Unmarshal(listMysqlAccountsRespJson, &listMysqlAccountsRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("users", flattenListAccountsBody(filterListAccounts(d, listMysqlAccountsRespBody))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListAccountsBody(resp []interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	rst := make([]interface{}, 0, len(resp))
+
+	for _, v := range resp {
+		rst = append(rst, map[string]interface{}{
+			"name":        utils.PathSearch("name", v, nil),
+			"hosts":       utils.PathSearch("hosts", v, nil),
+			"description": utils.PathSearch("comment", v, nil),
+		})
+	}
+	return rst
+}
+
+func filterListAccounts(d *schema.ResourceData, resp interface{}) []interface{} {
+	accountsJson := utils.PathSearch("users", resp, make([]interface{}, 0))
+	accountArray := accountsJson.([]interface{})
+	if len(accountArray) < 1 {
+		return nil
+	}
+	result := make([]interface{}, 0, len(accountArray))
+
+	rawName, rawNameOK := d.GetOk("name")
+	rawhost, rawhostOk := d.GetOk("host")
+
+	for _, account := range accountArray {
+		name := utils.PathSearch("name", account, nil)
+		if rawhostOk {
+			hostsJson := utils.PathSearch("hosts", account, make([]interface{}, 0))
+			hostArray := hostsJson.([]interface{})
+			for _, host := range hostArray {
+				if host == rawhost {
+					result = append(result, account)
+				}
+			}
+			continue
```

Comment:
this logic is error, it indicates that if the result contain the target host, then it will be returned